### PR TITLE
[libc] add proxy header for struct_sigaction

### PR DESCRIPTION
**Pull request by SchrodingerZhu**
Split from #91572

### CODE REVIEW

File: libc/src/signal/sigaction.h [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #ifndef LLVM_LIBC_SRC_SIGNAL_SIGACTION_H
 #define LLVM_LIBC_SRC_SIGNAL_SIGACTION_H
 
-#include <signal.h>
+#include <hdr/types/struct_sigaction.h>
```
Comment (michaelrj-google) [inline libc/src/signal/sigaction.h:12]: this should be `#include "hdr/types/struct_sigaction.h"`

In general `<angle brackets>` should only be used for public headers

**Review by michaelrj-google — APPROVED** [review]
LGTM after one change

**Review by nickdesaulniers — COMMENTED** [review]
Can we clean up test/src/signal/sigaltstack_test.cpp with this change?

and test/src/signal/sigaction_test.cpp
src/signal/linux/signal.cpp

File: libc/hdr/types/CMakeLists.txt [hunk 2]
```diff
@@ -126,3 +126,12 @@ add_proxy_header_library(
...
     libc.include.stdlib
 )
+
+add_proxy_header_library(
+  struct_sigaction
+  HDRS
+    struct_sigaction.h
+  FULL_BUILD_DEPENDS
+    libc.include.llvm-libc-types.struct_sigaction
+    libc.include.signal
```
Comment (nickdesaulniers) [inline libc/hdr/types/CMakeLists.txt:136]: Why do we need `libc.include.signal`?

Comment (SchrodingerZhu) [reply libc/hdr/types/CMakeLists.txt:136]: I think it is - convention in proxy headers to require the full build to generate the outer-level header. This is employed in all other proxy type headers. @lntue

**Review by nickdesaulniers — APPROVED** [review]

File: libc/hdr/types/CMakeLists.txt (hunk 2 above)
Comment (nickdesaulniers) [reply libc/hdr/types/CMakeLists.txt:136]: Approving for now, but perhaps worth revisiting this convention. @lntue ?

Comment (SchrodingerZhu) [reply libc/hdr/types/CMakeLists.txt:136]: merge this for now. if the convention is revised, we should better revisit all other files.

**Pull request merged**